src/main/java/org/terasology/structureTemplates/internal/systems/StructureSpawnServerSystem.java

```diff
@@ -167,8 +171,11 @@ public void onGetStructureTemplateBlocks(GetStructureTemplateBlocksEvent event,
         Map<Integer, List<BlockToPlace>> blocksPerLayer = Maps.newTreeMap();
         for (RegionToFill regionToFill : spawnBlockRegionComponent.regionsToFill) {
             Block block = regionToFill.blockType;
-
-            Region3i region = regionToFill.region;
+            if(entity.hasComponent(IgnoreAirBlocksComponent.class)&&block.getURI().getBlockFamilyDefinitionUrn().equals(BlockManager.AIR_ID.getBlockFamilyDefinitionUrn()))
+            {
+                continue;
+            }
+                Region3i region = regionToFill.region;
```

Comment:
This is the same block as above, which is always an indicator that you could do a refactoring to share the code for both usages. In this case, I would go for reusing the `isAir()` check but keep the general pattern in place here:

```suggestion
            if(entity.hasComponent(IgnoreAirBlocksComponent.class) && isAir(block)) {
                continue;
            }
            
            Region3i region = regionToFill.region;
```

Reply:
With generalised filters (sometime in the future) this may look like this

```java
   BlockFilterComponent filter = entity.getComponent(BlockFilterComponent.class);
   if (!skipBlock(block, filter)) {
     // ... code to fill block regions
   }         
```